### ISSUE TITLE
Impossible to troubleshoot if a metric switches from int to str (or vice versa)

### ISSUE THREAD

**Issue by ThomasBergman1** [indeedeng/imhotep#29]
- imhotep automatically detects and assigns type (int or str) to all fields in a shard based on the value of those fields
- if the type of a field switches (eg is sometimes a str and sometimes an int) imhotep has a bad time and that field cannot be used in the group-by
- however there is no way to know way to know which metrics have which types in which shards

It would be really nice to have some sort of a debug / validation tool that let's operators know which fields in which datasets contain multiple types, and for which shards.

Otherwise when this happens we basically have to backfill all time and hope it fixes it.

**Comment by ThomasBergman1** [comment]
To follow up on this, our big problems are as follows:

1) We have numerical metrics we would like to index, for which 'null' and '0' have semantically different meanings

eg.
'Users minimum price filter'
'displayed discount value' 
etc

2) We would like to be able to do large backfills of datasets without conflating 'null' and '0' concepts.
Eg if we begin tracking a new metric today, ideally we would not need to say that the value of the metric was '0' for all previous shards.

3) The rules for a field being an int are as follows (per the documentation):
"For Imhotep to treat a field’s value as an integer, at least 90% of the values must be integers or blanks, and at least 20% of the total values must be valid integers."

This poses a problem when we have a metric that is only true 20% of the time, OR, which used to be less common but has now become more common.

In this case the field can switch from int to str (or vice versa) and then become no longer usable in groupby.

----

In conclusion:
How are we supposed to handle null values for fields we want to be integers?
